File: src/main/java/com/flexpoker/web/commandsenders/InMemoryAsyncGameCommandSender.java
package com.flexpoker.web.commandsenders;

import com.flexpoker.framework.command.CommandReceiver;
import com.flexpoker.framework.command.CommandSender;
import com.flexpoker.game.command.commands.GameCommand;
import org.springframework.context.annotation.Lazy;
import org.springframework.stereotype.Component;

import javax.inject.Inject;

@Component
public class InMemoryAsyncGameCommandSender implements CommandSender<GameCommand> {

    private final CommandReceiver<GameCommand> gameCommandReceiver;

    @Lazy
    @Inject
    public InMemoryAsyncGameCommandSender(CommandReceiver<GameCommand> gameCommandReceiver) {
        this.gameCommandReceiver = gameCommandReceiver;
    }

    @Override
    public void send(GameCommand command) {
        gameCommandReceiver.receive(command);
    }

}

File: src/main/kotlin/com/flexpoker/web/commandsenders/InMemoryAsyncGameCommandSender.kt
package com.flexpoker.web.commandsenders

import com.flexpoker.framework.command.CommandReceiver
import com.flexpoker.framework.command.CommandSender
import com.flexpoker.game.command.commands.GameCommand
import org.springframework.context.annotation.Lazy
import org.springframework.stereotype.Component
import javax.inject.Inject

@Component
class InMemoryAsyncGameCommandSender @Lazy @Inject constructor(
    private val gameCommandReceiver: CommandReceiver<GameCommand>
) : CommandSender<GameCommand> {

    override fun send(command: GameCommand) {
        gameCommandReceiver.receive(command)
    }

}